Mathlib/GroupTheory/Congruence.lean

```diff
@@ -1257,6 +1258,18 @@ protected theorem div : ∀ {w x y z}, c w x → c y z → c (w / y) (x / z) :=
 #align con.div Con.div
 #align add_con.sub AddCon.sub
 
+/-- Multiplicative congruence relations preserve finite product. -/
+@[to_additive "Additive congruence relations preserve finite sum."]
+protected theorem finset_prod {ι M : Type*} [CommMonoid M] (c : Con M) (s : Finset ι)
+    {f g : ι → M} (h : ∀ i ∈ s, c (f i) (g i)) :
+    c (s.prod f) (s.prod g) := by
+  classical
+  induction s using Finset.induction_on with
```

Comment:
```suggestion
  induction s using Finset.cons_induction_on with
```
with a small tweak to the proof below.

Reply:
Done in [99fc44a](https://github.com/leanprover-community/mathlib4/pull/13401/commits/99fc44a9a0d81f339a69757d5edce5cb673fb936).

Why is `Finset.cons_induction_on` preferred over `Finset.induction_on` btw?